### ISSUE TITLE
Unnecessary trailing commas are no longer removed if an expression fits in one line. 

### ISSUE THREAD

**Issue by maximdeclercq** [psf/black#1699]
Unnecessary trailing commas should be removed if an expression fits in one line.
This doesn't happen for multiline lists e.g.

**To Reproduce**:

1.  Run _Black_ on the following code without arguments
```python
test = ["a very short line", "another element", "something long", "lorem ipsum dolor sit amet"]
```
2. _Black_ produces the following output as expected
```python
test = [
    "a very short line",
    "another element",
    "something long",
    "lorem ipsum dolor sit amet",
]
```
3. Change the last element to `"lorem ipsum",` and run _Black_ on it without arguments.
4. _Black_ changes nothing
```python
test = [
    "a very short line",
    "another element",
    "something long",
    "lorem ipsum",
]
```

**Expected behavior**
Running _Black_ on 
```python
test = [
    "a very short line",
    "another element",
    "something long",
    "lorem ipsum",
]
```
Should have the same result as running _Black_ on 
```python
test = [
    "a very short line",
    "another element",
    "something long",
    "lorem ipsum"
]
```
Output:
```python
test = ["a very short line", "another element", "something long", "lorem ipsum"]
```

**Environment:**
- Version:  20.8b1
- OS and Python version: Linux/Python 3.8.5

**Does this bug also happen on master?**
Yes, see this [link](https://black.now.sh/?version=master,&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AC9AH1dAD2IimZxl1N_WmhEpF7gpPYWgv97TAav44oRPN4fwYbEzwGqV7nJOvWw7PjG-p_RyfLIrLUpylKYXQJJz53De0Rv7xqc85nOotiIf26yyMQpo8PHAp-5LWFs07MnA8RhwLqqOs9Dg9GVBoCmKymmF6KE7lRdxlbvlRu0VY9iAAAAABNJk5TSCDgHAAGZAb4BAAB46TM4scRn-wIAAAAABFla).

**Comment by ichard26** [comment]
Black used to remove the trailing comma if the expression fits in a single line, but this was changed by #826 and #1288. Now a trailing comma tells Black to always explode the expression. This new feature / change was made mostly for the cases where you _know_ a collection or whatever will grow in the future. Having it always exploded as one element per line reduces diff noise when adding elements. Before the "magic trailing comma" feature, you couldn't anticipate a collection's growth reliably, which hurt Black's ability of reducing diff churn.

So no, this is not a bug, but an intended feature. The reason why you're filing this issue is probably since we say the following in the (outdated) style documentation:
> Unnecessary trailing commas are removed if an expression fits in one line. This makes it 1% more likely that your line won't exceed the allotted line length limit. Moreover, in this scenario, if you added another argument to your call, you'd probably fit it in the same line anyway. That doesn't make diffs any larger.

We missed that this paragraph became incorrect when the "magic trailing comma" feature was introduced. It was eventually fixed in commit 6b935a34d08c0b18d858e7da5093dc69a29c55f2, but that was after the stable documentation was released alongside `Black 20.8b1`. 

Anyway, [here's the documentation](https://github.com/psf/black/blob/master/docs/the_black_code_style.md#the-magic-trailing-comma) on the "magic trailing comma". Hopefully that helps and sorry for the possible confusion.

**Comment by zzzeek** [comment]
it would be super awesome if this were to be the second thing that black allows to be configurable.   because right now it's arbirary, if I have code like:

```
def x():
    foo = ('one', 'two', 'three')

    bar = ('one', 'two', 'three',)


```

now it becomes:

```
def x():
    foo = ("one", "two", "three")

    bar = (
        "one",
        "two",
        "three",
    )

```

that is terrible.   We now have to write a massive regexp because black refuses to do something that it's perfectly capable of and is making our life harder.
